### DISCUSSION TITLE
angle calculation query

### DISCUSSION THREAD

**Discussion by thgink** [mtex-toolbox/mtex#502] [Ask Anything] [answered]
Dear Developers:
When I try the crystallographic calculator built-in MTEX-5.3, it seemed that the angle calculation would not work in the normal way. For instance, in the FCC copper, the crystal direction [1 0 0] is perpendicular to [0 1 1], while the function _angle_ gives a result with "45" degree.
code:
cs=crystalSymmetry.load('Cu-Copper.cif')
angle(Miller(1,0,0,cs,'hkl'),Miller(0,1,1,cs,'uvw'))./degree
% 45
angle(Miller(1,0,0,cs,'uvw'),Miller(0,1,1,cs,'uvw'))./degree
% 45
![FCC Cu 100-011](https://user-images.githubusercontent.com/1176963/87305559-48082800-c549-11ea-839c-449b929df932.jpg)
![angle1](https://user-images.githubusercontent.com/1176963/87306931-58210700-c54b-11ea-84a0-2df0768730f6.jpg)
![angle2](https://user-images.githubusercontent.com/1176963/87306940-5c4d2480-c54b-11ea-9fc6-e296ba0283bc.jpg)

**Comment by kilir** [comment]
Hi,
angle takes always the smallerst angle between symmetrically equivalent directions.
Try:
```
angle(Miller(1,0,0,cs,'uvw'),Miller(0,1,1,cs,'uvw'),'noSymmetry')./degree
```

**Comment by thgink** [comment]
Hi,
"angle(Miller(1,0,0,cs,'uvw'),Miller(0,1,1,cs,'uvw'),'noSymmetry')./degree" do gives the right results, while in a deeper case, for instance in the function _orientation.map_,  still a relevant error pops out.

a=crystalSymmetry('m-3m',[5.9612,5.9612,5.9612],[90,90,90]*degree, 'X||a*','Z||c','mineral','Autensite');
m=crystalSymmetry('2',[4.5134,5.2117,38.2979],[90,87.5,90]*degree, 'X||a*','Z||c','mineral','Martensite');
misori = orientation.map(Miller(0,1,1,a,'hkl'),Miller(1,-1,1,a,'uvw'),Miller(-1,-2,8,m,'hkl'),Miller(-2,1,0,m,'uvw'));

![2020-07-13_210459](https://user-images.githubusercontent.com/1176963/87307964-16915b80-c54d-11ea-81c7-385417255052.jpg)
Here, (0,1,1)⊥[1,-1,1] in the autensite and (-1,-2,8)⊥[-2,1,0] in the martensite.

**Comment by kilir** [comment, marked as answer]
I think that is in one part the syntax you had in the map command and actually for me, your definition of m throws an error related to the alignment  - but I guess that's your * signs are missing:

```
a=crystalSymmetry('m-3m',[5.9612,5.9612,5.9612],[90,90,90]*degree, 'X||a','Z||c','mineral','Autensite');
m=crystalSymmetry('2',[4.5134,5.2117,38.2979],[90,87.5,90]*degree, 'X||a','Z||c*','mineral','Martensite');

angle(Miller(0,1,1,a,'hkl'),Miller(1,-1,1,a,'uvw'),'noSymmetry')/degree
angle(Miller(-1,-2,8,m,'hkl'),Miller(-2,1,0,m,'uvw'),'noSymmetry')/degree

misori = orientation.map(Miller(0,1,1,a,'hkl'),Miller(-1,-2,8,m,'hkl'),Miller(1,-1,1,a,'uvw'),Miller(-2,1,0,m,'uvw'));
```

**Comment by ralfHielscher** [comment]
Hi,

you are using the wrong syntax for orientation.map

Have a look here https://mtex-toolbox.github.io/RotationDefinition.html#11
for the correct ordering of the arguments





********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Comment by thgink** [comment]
Hi kilir and ralfHielscher,
Thanks very much for your patient answer!
